### DISCUSSION TITLE
Writer in Model to create objects

### DISCUSSION THREAD

**Discussion by andrepaulo-bit** [Nozbe/WatermelonDB#1263] [Q&A]
In: https://nozbe.github.io/WatermelonDB/Writers.html it says:
> We highly recommend defining writer methods on Models to organize all code that changes the database in one place, and only use inline writers sporadically.

How can I create the writer to create an object?

**Comment by fullstackNRJ** [comment]
Inside your model class just use @writer decorator on your async function.
`@writer 
   async addComment(body, author) {
    const newComment = await this.collections.get('comments').create(comment => {
      comment.post.set(this)
      comment.author.set(author)
      comment.body = body
    })
    return newComment
  }`

In you async function you get the table by passing name and call create method on it, this create method has a comment object which you can use to set or assign values to each of the required field, as you can see in the code.
Now, you can call `addComment` method directly from your model object.  Something like 
Comment.addComment(arg1, arg2)
